p2p/handshake/handshake.go

```diff
@@ -43,7 +43,8 @@ const (
 
 // HandshakeMessage is a handshake message.
 type HandshakeMessage struct {
-	GenesisID types.Hash20
+	GenesisID  types.Hash20
+	FirstLayer types.LayerID
```

Comment:
we can generalize it and define libp2p stream protocol that looks as /<genesis id>/layer/hs/1. then naturally if we update genesis id or layer id - peers won't be able to setup handshake